drivers/power/ltc3337/ltc3337.c

```diff
@@ -0,0 +1,659 @@
+{
+	uint16_t regValue;
+	uint8_t prescale;
+
+	int32_t ret;
+	assert(dev);
+
+	ret = ltc3337_read_reg(dev, LTC3337_REG_B, &regValue);
+
+	if( ret == 0 ) {
+		if( rawValue != NULL ) {
+			*rawValue = regValue;
+		}
+
+		if( value != NULL ) {
+			prescale = (dev->latchedRegA >> LTC3337_RA_PRESCALE_POS) &
+				   LTC3337_RA_PRESCALE_MSK;
+			ltc3337_charge_count_to_A(regValue, prescale,
+						  iPeakLookupTable[dev->iPeakLatched].qLsb_nA_hr,
+						  value);
+		}
+	}
+
+	return ret;
+}
+
+
+int ltc3337_set_prescaler(struct ltc3337_dev* dev, uint8_t prescale)
+{
+	uint16_t regValue;
+	int32_t ret;
+	assert(dev);
+
+	regValue = dev->latchedRegA;
+	regValue &= ~(LTC3337_RA_PRESCALE_MSK << LTC3337_RA_PRESCALE_POS);
+	regValue |= ((prescale & LTC3337_RA_PRESCALE_MSK) << LTC3337_RA_PRESCALE_POS);
+	ret = ltc3337_write_reg(dev, LTC3337_REG_A, regValue);
+	if( ret == 0 ) {
+		dev->latchedRegA = regValue;
+	}
+
+	return ret;
+}
+
+
+int ltc3337_set_temperature_alarms_C(struct ltc3337_dev* dev, int16_t hotAlarm,
+				     int16_t coldAlarm )
+{
+	uint16_t regVal;
+	uint8_t tempVal;
+
+	assert(dev);
+
+	if((hotAlarm < LTC3337_MIN_TEMP_C) || (hotAlarm > LTC3337_MAX_TEMP_C) ||
+	    (coldAlarm < LTC3337_MIN_TEMP_C) || (coldAlarm > LTC3337_MAX_TEMP_C)) {
+		return -1;
+	}
+
+	tempVal = ltc3337_temp_C_to_reg(hotAlarm);
+	regVal = (tempVal & LTC3337_RH_TEMP_MSK) << LTC3337_RH_HOT_ALARM_POS;
+	tempVal = ltc3337_temp_C_to_reg(coldAlarm);
+	regVal |= (tempVal & LTC3337_RH_TEMP_MSK) << LTC3337_RH_COLD_ALARM_POS;
+
+	return ltc3337_write_reg(dev, LTC3337_REG_H, regVal);
+}
+
+
+int ltc3337_set_counter_shutdown(struct ltc3337_dev* dev, uint8_t shutdownEn )
+{
+	uint16_t regValue;
+	int32_t ret;
+
+	assert(dev);
+
+	regValue = dev->latchedRegA;
+
+	if( shutdownEn ) {
+		regValue |= LTC3337_RA_SHTDN;
+	} else {
+		regValue &= ~(LTC3337_RA_SHTDN);
+	}
+
+	ret = ltc3337_write_reg(dev, LTC3337_REG_A, regValue);
+	if( ret == 0 ) {
+		dev->latchedRegA = regValue;
+	}
+
+	return ret;
+}
+
+
+int ltc3337_set_counter_alarm(struct ltc3337_dev* dev, uint16_t counterRegVal,
+			      uint8_t roundUp )
+{
+	uint8_t dataToSet;
+	uint16_t regValue;
+	int32_t ret;
+
+	assert(dev);
+
+	regValue = dev->latchedRegA;
+
+	//Only the upper 8 bits are utilized
+	dataToSet = (counterRegVal >> 8);
+
+	if(( roundUp ) && (counterRegVal & 0xFF) && (dataToSet != 0xFF)) {
+		dataToSet++;
+	}
+
+	regValue &= ~(LTC3337_RA_ALARM_LVL_MSK << LTC3337_RA_ALARM_LVL_POS);
+	regValue |= ((dataToSet & LTC3337_RA_ALARM_LVL_MSK) <<
+		     LTC3337_RA_ALARM_LVL_POS);
+
+	ret = ltc3337_write_reg(dev, LTC3337_REG_A, regValue);
+	if( ret == 0 ) {
+		dev->latchedRegA = regValue;
+	}
+
+	return ret;
+}
+
+
+int ltc3337_set_accumulated_charge(struct ltc3337_dev* dev, uint16_t regValue,
+				   uint8_t roundUp )
+{
+	uint16_t dataToSet;
+
+	assert(dev);
+
+	//Only upper 8-bits are writable
+	dataToSet = regValue & 0xFF00;
+
+	if(( roundUp ) && (regValue & 0xFF) && (dataToSet != 0xFF00)) {
+		dataToSet += 0x100;
```

Comment:
Avoid using magic numbers.

Reply:
Agree on magic numbers. This is just incrementing the MSB of a 16-bit word.  Creating a definition didn't seem value added, addressed the operation with code comments  instead.